packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-builder.ts

```diff
@@ -840,28 +855,43 @@ export class AmplifyS3ResourceCfnStack extends AmplifyResourceCfnStack implement
     return policy;
   }
 
-  //S3TriggerBucketPolicy - Policy to control trigger function access to S3 bucket
-  createTriggerPolicy(logicalId: string, triggerFunctionName: string): iamCdk.CfnPolicy {
-    let policyDefinition: IAmplifyPolicyDefinition = {
-      logicalId,
-      isPolicyNameAbsolute: true, // set if policyName is not a reference
-      policyNameRef: 'amplify-lambda-execution-policy-storage',
-      roleRefs: [`function${triggerFunctionName}LambdaExecutionRole`],
-      statements: [
-        {
-          refStr: 'S3Bucket',
-          pathStr: '/*',
-          isActionAbsolute: true, //actions are not refs
-          actions: ['s3:PutObject', 's3:GetObject', 's3:ListBucket', 's3:DeleteObject'],
-          effect: iamCdk.Effect.ALLOW,
-        },
+  //helper: builder for policy statement resource
+  buildResourceFromPolicyDefinition(policyDefinitionStatement : IAmplifyIamPolicyStatementParams ){
+    let resourceStrArr = ['arn:aws:s3:::', { Ref: policyDefinitionStatement.refStr, } ];
+    if ( policyDefinitionStatement.pathStr && policyDefinitionStatement.pathStr.length > 0 ){
```

Comment:
nit:
```suggestion
    if (policyDefinitionStatement.pathStr?.length > 0 ){
```

Reply:
This was done to address a Typescript warning.